apps/api/src/config/validation.config.ts

```diff
@@ -9,6 +9,7 @@ const globalValidationPipe = new ValidationPipe({
   transformOptions: {
     strategy: 'exposeAll',
     excludeExtraneousValues: true,
+    exposeDefaultValues: true,
```

Comment:
what is the purpuse of this

Reply:
If a field inside the dto is initialized , the field with the initialized value will be "attached" to the request body, if that field was not provided in the request.

Without setting this property to true, the initialized field from [CreateAffiliateDonationDto](https://github.com/podkrepi-bg/api/pull/564/files#diff-f6b5c69a98f12a5d976a182325ca630f28c638f37d797bb95194f8b19a9397f4) will be undefined, if not provided.

Though looking at it more clearly now, I should remove the validator decorators from `affiliateId` and leave it just as a type, as it something we should receive from the affiliateCode, rather than it being part of the request.

Reply:
Yes, you are right. BEcause you get the affiliate document from the code. When you say initialized do you mean the fields where you use this pattern `string | null = null`. My only concern here is since the option is a global and affects all dto's, that  it won't create unexpected results in other routes. But I suppose this is the first time this initialization pattern is being used, so should not be a problem